src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c

```diff
@@ -22,23 +38,108 @@ const ActorInit Bg_Zg_InitVars =
     (ActorFunc)BgZg_Update,
     (ActorFunc)BgZg_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Zg/BgZg_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Zg/func_808C0C50.s")
+extern u32 D_06001080;
+extern u32 D_060011D4;
+
+static void BgZg_Destroy(BgZg* this, GlobalContext* globalCtx) {
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+static void func_808C0C50(BgZg* this) {
+    Audio_PlaySoundGeneral(NA_SE_EV_METALDOOR_OPEN, &this->dyna.actor.unk_E4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+}
+
+static s32 func_808C0C98(BgZg* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+    s32 flag;
+    flag = (thisx->params >> 8) & 0xFF;
```

Comment:
This is really minor but I would have the flag assignment on the same line if you're doing it for thisx.